diff --git b/api/webui/src/apis/files_api.rs a/api/webui/src/apis/files_api.rs
index 229bcaf..8d79752 100644
--- b/api/webui/src/apis/files_api.rs
+++ a/api/webui/src/apis/files_api.rs
@@ -213,6 +213,7 @@ pub async fn get_file_by_id_files_id_get(
 pub async fn get_file_content_by_id_files_id_content_file_name_get(
     configuration: &configuration::Configuration,
     id: &str,
+    file_name: &str,
 ) -> Result<models::FileModel, Error<GetFileContentByIdFilesIdContentFileNameGetError>> {
     let local_var_configuration = configuration;
 
@@ -446,8 +447,9 @@ pub async fn upload_file_files_post(
     if let Some(ref local_var_token) = local_var_configuration.bearer_access_token {
         local_var_req_builder = local_var_req_builder.bearer_auth(local_var_token.to_owned());
     };
-    let mut local_var_form = reqwest::multipart::Form::new();
-    // TODO: support file upload for 'file' parameter
+
+    let local_var_form = reqwest::multipart::Form::new().file("file", file).await?;
+
     local_var_req_builder = local_var_req_builder.multipart(local_var_form);
 
     let local_var_req = local_var_req_builder.build()?;
